Lib/operator.py

```diff
@@ -80,6 +80,16 @@ def and_(a, b):
     "Same as a & b."
     return a & b
 
+def as_float(obj):
+    """
+    Convert something numeric to float.
+
+    Same as float(obj), but does not accept strings.
+    """
+    if isinstance(obj, (str, bytes, bytearray)):
```

Comment:
For a better match with the C code, this also needs to exclude buffer objects. I'm not sure what the right way to do that is.

Reply:
Update to try converting to memoryview. If the conversion succeeds, we have a buffer object.

That's not quite right, because we could still have something that supports the buffer protocol but _also_ implements __float__ or __index__. But maybe this is close enough.